src/com/google/javascript/jscomp/deps/ModuleLoader.java

```diff
@@ -149,32 +157,80 @@ public String toModuleName() {
     }
 
     /**
-     * Find a CommonJS module {@code requireName}. See
+     * Find a JS module {@code requireName}. See
+     * https://nodejs.org/api/modules.html#modules_all_together
+     *
+     * @return The normalized module URI, or {@code null} if not found.
+     */
+    @Nullable
+    public ModulePath resolveJsModule(String moduleAddress) {
+      return resolveJsModule(moduleAddress, null, -1, -1);
+    }
+
+    /**
+     * Find a JS module {@code requireName}. See
      * https://nodejs.org/api/modules.html#modules_all_together
      *
      * @return The normalized module URI, or {@code null} if not found.
      */
-    public ModulePath resolveCommonJsModule(String requireName) {
-      String loadAddress;
+    @Nullable
+    public ModulePath resolveJsModule(
+        String moduleAddress, String sourcename, int lineno, int colno) {
+      String loadAddress = null;
 
       // * the immediate name require'd
-      if (isAbsoluteIdentifier(requireName) || isRelativeIdentifier(requireName)) {
-        loadAddress = resolveCommonJsModuleFileOrDirectory(requireName);
-      } else {
-        loadAddress = resolveCommonJsModuleFromRegistry(requireName);
+      switch (resolutionMode) {
+        case LEGACY:
+          loadAddress = resolveJsModuleFile(moduleAddress);
```

Comment:
@ChadKillingsworth 
The only breakages I know of are the 2 related to closure rules I mentioned above.
I'm not sure of exactly why they fail,
but the LEGACY code path is definitely not defining the same behavior as before this PR,
so I believe the answer lies somewhere in that difference.

Since you said you updated it, I'll grab a fresh copy of this PR.
I'm going to modify it a bit before submission to ensure that the LEGACY code path preserves current behavior.